src/app/starredentries/starredentries.component.html

```diff
@@ -73,12 +73,7 @@
             </mat-card>
             <div fxLayout="column" fxLayoutGap="1rem" class="mb-2 mx-1">
               <mat-card fxFlex *ngFor="let tool of starredTools; let last = last" class="m-1 pb-3">
-                <ng-container
-                  *ngTemplateOutlet="
-                    tool.mode === 'DOCKSTORE_YML' ? workflowSummary : toolSummary;
-                    context: tool.mode === 'DOCKSTORE_YML' ? { workflow: tool, entryType: entryType.AppTool } : { tool: tool }
-                  "
```

Comment:
This seems like app tool specific code, what was the reason for removing this?

Reply:
> This seems like app tool specific code, what was the reason for removing this?

Before the change, the compiler was returning this error with strictTemplates turned on:

```
Error: src/app/starredentries/starredentries.component.html:79:30 - error TS2367: This condition will always return 'false' since the types 'ModeEnum' and '"DOCKSTORE_YML"' have no overlap.

79                     context: tool.mode === 'DOCKSTORE_YML' ? { workflow: tool, entryType: entryType.AppTool } : { tool: tool }
                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```

IOW, it always evaluated to false, because DockstoreTool.ModeEnum is defined as `export type ModeEnum = 'AUTO_DETECT_QUAY_TAGS_AUTOMATED_BUILDS' | 'AUTO_DETECT_QUAY_TAGS_WITH_MIXED' | 'MANUAL_IMAGE_PATH' | 'HOSTED';`; there is no `DOCKSTORE_YML`.

So I just  narrowed the code down to the `false` condition is was always evaluating to. But my approach may have just perpetuated a bug; will investigate.

Reply:
That was a good catch, updated the code.

The issue was that we were casting `Entry` to `DockstoreExtendedTool` in the controller, and `DockstoreExtendedTool` doesn't have a mode that can be `DOCKSTORE_YML`.

The cast was incorrect, as the endpoint can return AppTools, so I changed the type of starredEntries to `starredTools: Array<ExtendedDockstoreTool | ExtendedWorkflow> | null;`.

I then changed the HTML template to use the `entryType` property, which is more obvious than the mode.